src/Time.svelte

```diff
@@ -1,67 +1,74 @@
 
-  /**
-   * Original timestamp
-   * @type {import("dayjs").ConfigType}
-   */
-  export let timestamp = new Date().toISOString();
+  const {
+    /**
+     * Original timestamp
+     * @type {import("dayjs").ConfigType}
+     */
+    timestamp = new Date().toISOString(),
 
-  /**
-   * Timestamp format for display.
-   * It's also used as a title in the `relative` mode
-   * @type {import("dayjs").OptionType}
-   * @example "YYYY-MM-DD"
-   */
-  export let format = "MMM DD, YYYY";
+    /**
+     * Timestamp format for display.
+     * It's also used as a title in the `relative` mode
+     * @type {import("dayjs").OptionType}
+     * @example "YYYY-MM-DD"
+     */
+    format = "MMM DD, YYYY",
 
-  /**
-   * Set to `true` to display the relative time from the provided `timestamp`.
-   * The value is displayed in a human-readable, relative format (e.g., "4 days ago", "Last week")
-   * @type {boolean}
-   */
-  export let relative = false;
+    /**
+     * Set to `true` to display the relative time from the provided `timestamp`.
+     * The value is displayed in a human-readable, relative format (e.g., "4 days ago", "Last week")
+     * @type {boolean}
+     */
+    relative = false,
 
-  /**
-   * Set to `true` to update the relative time at 60 second interval.
-   * Pass in a number (ms) to specify the interval length
-   * @type {boolean | number}
-   */
-  export let live = false;
-
-  /**
-   * Formatted timestamp.
-   * Result of invoking `dayjs().format()`
-   * @type {string}
-   */
-  export let formatted = "";
+    /**
+     * Set to `true` to update the relative time at 60 second interval.
+     * Pass in a number (ms) to specify the interval length
+     * @type {boolean | number}
+     */
+    live = false,
+    ...rest
+  } = $props();
 
   import { dayjs } from "./dayjs";
   import { onMount } from "svelte";
 
   /** @type {undefined | NodeJS.Timeout} */
   let interval = undefined;
+  let liveUpdate = $state(0);
 
   const DEFAULT_INTERVAL = 60 * 1_000;
 
   onMount(() => {
     return () => clearInterval(interval);
   });
 
-  $: if (relative && live !== false) {
-    interval = setInterval(
-      () => {
-        formatted = dayjs(timestamp).from();
-      },
-      Math.abs(typeof live === "number" ? live : DEFAULT_INTERVAL),
-    );
-  }
-  $: formatted = relative
-    ? dayjs(timestamp).from()
-    : dayjs(timestamp).format(format);
-  $: title = relative ? dayjs(timestamp).format(format) : undefined;
+  $effect(() => {
+    if (relative && live !== false) {
+      interval = setInterval(
+        () => ++liveUpdate,
+        Math.abs(typeof live === "number" ? live : DEFAULT_INTERVAL),
+      );
+    }
+  });
+
+  /**
+   * Formatted timestamp.
+   * Result of invoking `dayjs().format()`
+   * @type {string}
+   */
+  let formatted = $derived.by(() => {
+    liveUpdate; // no-op. this is a dependency trigger for live updates
+    return relative ? dayjs(timestamp).from() : dayjs(timestamp).format(format);
+  });
```

Comment:
I don't think that we need the `liveUpdate` $state rune b/c if any of the props change, `formatted` should get updated (i think...). it is something i can test. if not, this would simplify some things.

Reply:
i have removed liveUpdate as it does not have any adverse impacts on live updates.